### PR TITLE
Section 2: Intro to ruby

### PR DESCRIPTION
**Pull request by ziqiang2344**
This PR is stacking on top of [Section 1: Let's code the quiz!](https://github.com/ziqiang2344/Learn_Ruby_on_Rails/pull/1) PR and this PR is stacked by [Section 2: Working with Strings PR](https://github.com/ziqiang2344/Learn_Ruby_on_Rails/pull/3). 

Hence, [Section 1: Let's code the quiz PR](https://github.com/ziqiang2344/Learn_Ruby_on_Rails/pull/1) has to be merged before this PR and this PR has to be merged before [Section 2: Working with String PR](https://github.com/ziqiang2344/Learn_Ruby_on_Rails/pull/3).

---
[If Kiong were to do it]

- https://github.com/ziqiang2344/Learn_Ruby_on_Rails/pull/1 needs to be merged before this PR can be merged

### CODE REVIEW

**Comment by Michelle-Lohwt** [comment]
Change your indentation to 2 ya

File: sec2_the_ruby_programming_language/2.1_output.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#creating variable
+greeting = "Hello World"
+
+puts greeting
+
+#----------------------------------------------------------------------
+
+#defining a method
+def say_hello
+    puts "Hello World"
```
Comment (Michelle-Lohwt) [inline sec2_the_ruby_programming_language/2.1_output.rb:30]: Change your indentation to 2 ya

**Review by Michelle-Lohwt — CHANGES_REQUESTED** [review]
Change your indentation to 2 ya

**Comment by ziqiang2344** [comment]
> Change your indentation to 2 ya
image
@Michelle-Lohwt do you mean by changing indentation to 2 from here?

**Review by tlkiong — CHANGES_REQUESTED** [review]
image
Please indicate if need to merge previous PR first

---

image

Click here if you need review again

---

Its not your editor. Its the code here.
Ensure its:
- 2 SPACES. not tabs

**Review by tlkiong — CHANGES_REQUESTED** [review]
image

Your comment is assuming that everyone here (including yourself) knows which is the PR you are referring to. Lets say you know and you have 1000 PRs closed in this repo now. YOu would have to sift through or search for it. Why not just simplify everything by linking it directly?

*Here uses [markdown](https://www.markdownguide.org/cheat-sheet/)

**Comment by tlkiong** [comment]
See my update on the description. 
I'll review this after the dependent PR is merged

**Review by Michelle-Lohwt — CHANGES_REQUESTED** [review]
Some of the files still have 4 indentation

File: sec1_intro_and_setup/1.1_basics.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+address = [1,2,3,4,5,6,7,8,9]
+
+#printing array
+p address
+
+#printing array with index 5
+p address[5]
+
+#assigning new_address to store the reverse of the array
+new_address = address.reverse!
```
Comment (tlkiong) [inline sec1_intro_and_setup/1.1_basics.rb]: ???

**Review by tlkiong — CHANGES_REQUESTED** [review]
Please double check your work.
Either you miss it or didn't update your branch with main

File: sec1_intro_and_setup/1.1_basics.rb (hunk 2 above)
Comment (ziqiang2344) [reply sec1_intro_and_setup/1.1_basics.rb]: I didn't update the branch with main. Now it's updated.

**Review by tlkiong — APPROVED** [review]
This is why we have the links. So, you can check after you merge one PR.
Always update your PR.

A way you can make sure you don't forget is in the settings of the repo, check the option to always keep the PR updated with main

**Pull request merged**